packages/web-component/src/lib/descope-wc/BaseDescopeWc.ts

```diff
@@ -488,66 +489,96 @@ class BaseDescopeWc extends HTMLElement {
-      return;
-    }
+    if (version) return version;
 
-    const existingScript = document.getElementById(
-      SCRIPT_ID,
-    ) as HTMLScriptElement;
-    const scriptEle: HTMLScriptElement =
-      existingScript || document.createElement('script');
+    this.logger.error('Did not get components version, using latest version');
 
-    if (existingScript) {
+    return 'latest';
+  }
+
+  static descopeUI: any;
+
+  async #loadDescopeUI() {
+    if (BaseDescopeWc.descopeUI) {
       this.loggerWrapper.info(
-        'DescopeUI loading script is already exist, probably multiple flows are running on the same page',
+        'DescopeUI is already loading, probably multiple flows are running on the same page',
       );
-    } else {
-      scriptEle.id = SCRIPT_ID;
-
-      let version = (await this.#getConfig())?.projectConfig?.componentsVersion;
+      return;
+    }
 
-      if (!version) {
-        this.logger.error(
-          'Did not get components version, using latest version',
-        );
-        version = 'latest';
+    BaseDescopeWc.descopeUI = new Promise((resolve) => {
+      if (globalThis.DescopeUI) {
+        resolve(globalThis.DescopeUI);
+        return;
       }
 
-      scriptEle.src = UI_COMPONENTS_URL.replace(
-        UI_COMPONENTS_URL_VERSION_PLACEHOLDER,
-        version,
-      );
+      const setupScript = (url: string) => {
+        const scriptEle = document.createElement('script');
+        scriptEle.id = 'load-descope-ui';
+        scriptEle.src = url;
 
-      document.body.append(scriptEle);
-    }
+        return scriptEle;
+      };
 
-    this.descopeUI = new Promise((res) => {
-      const onError = () => {
-        this.loggerWrapper.error(
-          'Cannot load DescopeUI',
-          `Make sure this URL is valid and return the correct script: "${scriptEle.src}"`,
+      const generateScriptUrl = (
+        urlTemplate: string,
+        componentsVersion: string,
+      ) =>
+        urlTemplate.replace(
+          UI_COMPONENTS_URL_VERSION_PLACEHOLDER,
+          componentsVersion,
         );
 
-        res(undefined);
+      const attachEvents = (
+        scriptEle: HTMLScriptElement,
+        onError: () => void,
+      ) => {
+        const onErrorInternal = () => {
+          this.loggerWrapper.error(
+            'Cannot load DescopeUI',
+            `Make sure this URL is valid and return the correct script: "${scriptEle.src}"`,
+          );
+
+          onError();
+        };
+
+        scriptEle.addEventListener('load', () => {
+          if (!globalThis.DescopeUI) onErrorInternal();
+          resolve(globalThis.DescopeUI);
+        });
+
+        scriptEle.addEventListener('error', onErrorInternal);
       };
 
-      scriptEle.addEventListener('load', () => {
-        if (!globalThis.DescopeUI) onError();
-        res(globalThis.DescopeUI);
-      });
+      (async () => {
+        const componentsVersion = await this.#getComponentsVersion();
```

Comment:
question about the `this` here, this line is defined as a static function, so it gets the value from the caller, is that right?

maybe worth passing it as an argument instead the implicit implicatio of the caller having this function defined 

not critical, lmk what you think

Reply:
Not sure I understand what do you mean by `it gets the value from the caller`

Basically we are initializing a static member (so we can share it across different instances of the web-component)

The init process is using non static methods (because there is no need to share it with other instances)

Do you suggest to change all the helper methods to static as well?

Reply:
But there is no static function here,
It's a static attribute

I suggest that we will have a short zoom call to close this

WDYT?